### DISCUSSION TITLE
posts/Go/priority_in_go_select/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#327] [Announcements]
https://www.liwenzhou.com/posts/Go/priority_in_go_select/ 

 李文周的Blog select 优先级 case default priority k8s 源码

**Comment by olonglongo** [comment]
老铁，这个MENU酷酷的。

**Comment by ldrmcml** [comment]
以下不会死循环，另外如何实现多个优先级chan？
https://studygolang.com/articles/6364
for {
    select {
    case data := <- highChan:
        handleHigh(data)
    default:
        select {
        case data := <- highChan:
            handleHigh(data)
        case data := <- lowChan:
            handleLow(data)
        }
    }
}

**Comment by Q1mi** [comment]
> @ldrmcml
> 以下不会死循环，另外如何实现多个优先级chan？
> https://studygolang.com/articles/6364
> for {
>     select {
>     case data := <- highChan:
>         handleHigh(data)
>     default:
>         select {
>         case data := <- highChan:
>             handleHigh(data)
>         case data := <- lowChan:
>             handleLow(data)
>         }
>     }
> }

select语言中当多个case语句同时满足时会随机选择一个执行，也就是你贴的这段代码第二层的select中不能保证`<-highChan`优先于`<-lowChan`执行。

**Comment by GGGIsGGG** [comment]
悄悄问一下，这里这个LABEL是不是不加也可以啊？只有1层循环诶，break直接就出来了。

**Comment by yin1999** [comment]
> @GGGIsGGG
> 悄悄问一下，这里这个LABEL是不是不加也可以啊？只有1层循环诶，break直接就出来了。

不可以，因为`select case`语句也是有`break`的，所以在里面写break是只能`break` `select case`，而不能把外层的`for`给`break`了

**Comment by bodhiye** [comment]
学习了

**Comment by Anaethesia** [comment]
很棒，有学到了。

**Comment by SeibertronSS** [comment]
```
for {
	select {
	case <-stopCh:
		return
```
这种跳出方式，意思是会在`stopCh`有值，需要 stop 的时候可能也会执行`job1` 和 `job2`吗？

**Comment by Pitfall-cmd** [comment]
我觉得我这个也一样可以把，
func worker2(ch1,ch2 <-chan int, stopCh chan struct{}){
	for{
		select {
		case <-stopCh:
			return
		case job1:=<-ch1:
			fmt.Println(job1)
		case job2:=<-ch2:
			select{
				case job1 := <-ch1:
					fmt.Println(job1)
				default:
				}
			fmt.Println(job2)
		}
	}
}

**Comment by Anaethesia** [comment]
您好，您的邮件已收到。
